vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/treegrid/it/TreeGridChangingHierarchyIT.java

```diff
@@ -138,6 +138,23 @@ public void remove_item_from_root() {
         Assert.assertEquals("b", grid.getCell(0, 0).getText());
     }
 
+    @Test
+    public void addChildrenToExpandedNode_childrenDisplayed() {
```

Comment:
The issue is fixed on the root level, but it still remains in the sub-levels:

https://github.com/vaadin/flow-components/assets/1222264/ce79ed3f-9f24-4622-8158-9f93a729011e

Reply:
Sure, done.